Fix trigger paths filters to execute precommit tests only if needed

iemejia opened the pull request:
I noticed that too many things were being executed as part of a PR in the Spark runner in particular Flink related validates runner and other tests, so I tried the matching patterns a bit.
Notice that for Python I removed it because it is already matched by the PortablePython precommit.

R: @angoenka @lukecwik 
CC: @mxm

iemejia commented:
Run Seed Job

iemejia commented:
PTAL @angoenka

iemejia commented:
@lukecwik added you as a reviewer on this one since I suppose @angoenka will be a bit busy with the ongoing release. PTAL.

angoenka commented on .test-infra/jenkins/job_PreCommit_Python_ValidatesRunner_Flink.groovy:
```diff
@@ -21,16 +20,19 @@ import PrecommitJobBuilder
+    triggerPathPatterns: [
+      '^model/.*$',
+      '^runners/core-construction-java/.*$',
+      '^runners/core-java/.*$',
```
We should also add 
`'^runners/extensions-java/.*$'`
`'^runners/reference/.*$'`

iemejia replied:
Why the reference runner should trigger a Flink build I don't see any relation between it and the Flink runner?

iemejia replied:
Oh I see now, it is used for Flink's Job server, shouldn't we get this code out of Reference now that it is getting deprecated?

iemejia replied:
done

angoenka commented on .test-infra/jenkins/job_PreCommit_Portable_Python.groovy:
```diff
@@ -24,7 +24,10 @@ PrecommitJobBuilder builder = new PrecommitJobBuilder(
-      '^runners/.*$',
+      '^runners/core-construction-java/.*$',
+      '^runners/core-java/.*$',
+      '^runners/flink/.*$',
```
We should also add 
`'^runners/extensions-java/.*$'`
`'^runners/reference/.*$'`

runner/reference contains PortableRunner used to submit pipelines.
runners/extensions-java contains metrics

iemejia replied:
done

iemejia commented:
Updated and rebased. Minor question: Shouldn't we move the parts of ReferenceRunner used by Flink and other runners out of the reference package now that it is getting deprecated?

angoenka commented:
Yes, We should move it out. I will take a shot at it later this week. 
Jira for tracking https://issues.apache.org/jira/browse/BEAM-7303

mxm commented:
We should run the seed job before merging this. There were some syntax errors in the Jenkins Job dsl scripts.

iemejia commented:
Oups we have done for the first version but I forgot to do it after fixing the comments. Is anything related to this breaking at the moment?

mxm commented:
All good. This has already been fixed: 26583d6871847b8297ab5c2a770c1939d1d74036

iemejia commented:
Thanks, sorry for that one, a silly mistake.

iemejia commented:
Thanks @adude3141 and sorry if the noise this generated hit on you.

adude3141 commented:
Do not worry, @iemejia. This was highly hidden by myself messing up test-infra. If I did not miss anything, this got never seeded anyway :)
